File: tests/fixtures/lang-coverage/sample.go
package main

import (
	"fmt"
	"os"
)

type Server struct {
	host string
	port int
}

func NewServer(host string, port int) *Server {
	return &Server{host: host, port: port}
}

func (s *Server) Start() error {
	fmt.Printf("Starting on %s:%d\n", s.host, s.port)
	return nil
}

func main() {
	srv := NewServer("localhost", 8080)
	if err := srv.Start(); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
}

File: tests/fixtures/lang-coverage/sample.cpp
#include <iostream>
#include <string>
#include <vector>

namespace umbra {

struct Config {
  std::string name;
  int port;
};

class Server {
public:
  explicit Server(Config cfg) : config_(cfg) {}

  void start() {
    std::cout << "Starting server: " << config_.name << std::endl;
  }

private:
  Config config_;
};

enum class Status { OK, Error, Pending };

template <typename T>
T clamp(T value, T min, T max) {
  if (value < min) return min;
  if (value > max) return max;
  return value;
}

} // namespace umbra

int main() {
  umbra::Config cfg{"umbra", 8080};
  umbra::Server server(cfg);
  server.start();
  return 0;
}
